internal/cli/plugin/plugin_github_asset_manager.go

```diff
@@ -43,55 +42,63 @@ var (
 	errCreatePluginZipFile          = errors.New("could not create plugin zip file")
 	errSaveAssetToPluginDir         = errors.New("failed to save asset to plugin directory")
 	errCreateDirToExtractAssetFiles = errors.New("failed to create to plugin directory to extract assets in")
+	errCreatePluginAssetFromPlugin  = errors.New("failed to create plugin asset from plugin")
 )
 
-type GithubRelease struct {
-	owner   string
-	name    string
-	version *semver.Version
+type GithubAssetManager struct {
+	ghClient *github.Client
+	owner    string
+	name     string
+	version  *semver.Version
 }
 
-type AssetOpts struct {
-	existingCommands []*cobra.Command
-	ghClient         *github.Client
-	githubRelease    *GithubRelease
+func (g *GithubAssetManager) repository() string {
+	return fmt.Sprintf("%s/%s", g.owner, g.name)
 }
 
-func (opts *AssetOpts) repository() string {
-	if opts.githubRelease == nil {
-		return ""
+func createGithubAssetManagerFromPlugin(p *plugin.Plugin) (*GithubAssetManager, error) {
+	if !p.HasGithub() {
+		return nil, errCreatePluginAssetFromPlugin
 	}
-	return fmt.Sprintf("%s/%s", opts.githubRelease.owner, opts.githubRelease.name)
+
+	return &GithubAssetManager{
+		owner: p.Github.Owner,
+		name:  p.Github.Name,
+	}, nil
+}
+
+func (g *GithubAssetManager) getPluginDirectoryName() string {
+	return fmt.Sprintf("%s@%s", g.owner, g.name)
 }
 
-func (opts *AssetOpts) getPluginAssetInfo() ([]*github.ReleaseAsset, error) {
+func (g *GithubAssetManager) getPluginAssetInfo() ([]*github.ReleaseAsset, error) {
```

Comment:
Would it make more sense to call this something along the lines of `getReleaseAssets`?
- It returns an array, but your function name is singular
- It returns some struct called `ReleaseAsset`

Reply:
done